### PR TITLE
CLDC-1903 Infer first time property let question

### PR DESCRIPTION
**Pull request by ghost**
To backfill the data, we will need to run the following script after deploy:

```
    LettingsLog.where(renewal: 1).update_all(first_time_property_let_as_social_housing: 0)
```

### CODE REVIEW

File: spec/models/lettings_log_spec.rb [hunk 1]
```diff
@@ -1454,6 +1454,15 @@ def previous_postcode_fields
         expect(record_from_db["vacdays"]).to eq(0)
         expect(lettings_log["vacdays"]).to eq(0)
       end
+
+      it "correctly derives and saves first_time_property_let_as_social_housing" do
+        record_from_db = ActiveRecord::Base.connection.execute(
```
Comment (asmega) [inline spec/models/lettings_log_spec.rb:1459]: why do we hand craft SQL queries when we have active record?

```suggestion
record_from_db = LettingsLog.find(id: lettings_log.id)
```

File: spec/models/lettings_log_spec.rb [hunk 2]
```diff
@@ -1454,6 +1454,15 @@ def previous_postcode_fields
...
+          "select first_time_property_let_as_social_housing" \
+          " from lettings_logs where id=#{lettings_log.id}",
+        ).to_a[0]
```
Comment (asmega) [inline spec/models/lettings_log_spec.rb:1462]: why do we hand craft SQL queries when we have active record?

```suggestion
        record_from_db = LettingsLog.find(lettings_log.id)
```

Comment (ghost) [reply spec/models/lettings_log_spec.rb:1462]: By looking at the rest of the file and what it is testing, I _think_ this is to bypass the ActiveRecord model and retrieve data straight from the database.

Comment (asmega) [reply spec/models/lettings_log_spec.rb:1462]: you can add a `select` if you really want to restrict the data we request and that comes back

Comment (asmega) [reply spec/models/lettings_log_spec.rb:1462]: well at this moment in time i'm gonna say the rest of the file is wrong. we don't interface with the database directly with SQL queries it goes through active record. therefore validating the database has the correct data with SQL, only to then interface with it through active record seems absurd

Comment (ghost) [reply spec/models/lettings_log_spec.rb:1462]: I totally agree with that. I will create a tech debt ticket to refactor this test file 👍

Comment (ghost) [reply spec/models/lettings_log_spec.rb:1462]: https://digital.dclg.gov.uk/jira/browse/CLDC-2033

**Review by natdeanlewissoftwire — APPROVED** [review]

File: spec/models/lettings_log_spec.rb (hunk 2 above)
Comment (jamesrose) [reply spec/models/lettings_log_spec.rb:1462]: Agreed!

Comment (kosiakkatrina) [reply spec/models/lettings_log_spec.rb:1462]: Context to why it is the way it is: 
At some point we weren't saving the inferred values to the db, but we had methods that would return that value on the model. After that we made the decision that we actually wanted to persist the value to the db. This code was a consequence of basically testing that.
If we had it as `expect(LettingsLog.find(lettings_log.id).field_we_are_testing_for).to eq(expected_value)` it wold call the method on the model and wouldn't fail as expected
I don't love how this test looks, but in terms of confidence I think it is testing the fact that the data is actually persisted and I'm not sure how else we could do that

File: app/models/derived_variables/lettings_log_variables.rb [hunk 3]
```diff
@@ -60,7 +62,6 @@ def set_derived_fields!
     if is_supported_housing? && location
       self.wchair = location.mobility_type_before_type_cast == "W" ? 1 : 2
     end
-    self.voiddate = startdate if is_renewal?
```
Comment (kosiakkatrina) [inline app/models/derived_variables/lettings_log_variables.rb:64]: nice

**Review by kosiakkatrina — APPROVED** [review]

File: spec/models/lettings_log_spec.rb (hunk 2 above)
Comment (asmega) [reply spec/models/lettings_log_spec.rb:1462]: We have a ticket now so at least someone can take a look at it. i think at least a few of us thing its less than ideal

**Pull request merged**